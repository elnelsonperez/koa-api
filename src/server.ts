import CreateApplication from '@app/app';
import { areWeTestingWithJest } from './helpers';
import usersSeeder from '@src/database/seeders/users.seeder';

const PORT:number = Number(process.env.PORT) || 3000;

CreateApplication().then(async (app) => {
    await usersSeeder();
    app.listen(PORT);
    if (!areWeTestingWithJest()) {
        console.log(`App Started at http://127.0.0.1:${PORT}`);
    }
});
