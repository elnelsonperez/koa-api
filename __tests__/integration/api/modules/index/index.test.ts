import * as request from 'supertest';
import server from '@src/server';

afterAll(() => {
    server.close();
});

describe('Index GET / ', () => {
    it('gets correct index api response', async () => {
        const response = await request(server).get('/');
        expect(response.status).toBe(200);
        expect(response.text).toMatchSnapshot();
    });
});
