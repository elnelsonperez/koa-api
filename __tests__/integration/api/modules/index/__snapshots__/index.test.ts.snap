// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index GET /  gets correct index api response 1`] = `"Koa API!"`;
